### PR TITLE
feat: add pdu interface and pdu header

### PR DESCRIPTION
**Pull request by uncommon-nickname**

### CODE REVIEW

File: src/pdu_header.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                         destinationEntityIDBytes.end());
+
+    return encodedHeader;
+};
+
+template <class EntityIDType, class SequenceNumberType>
+    requires std::unsigned_integral<EntityIDType> && std::unsigned_integral<SequenceNumberType>
+uint8_t cfdp::pdu::PduHeader<EntityIDType, SequenceNumberType>::buildFirstByte() const
+{
+    return (version << VERSION_POSITION) | (utils::toUnderlying(pduType) << 4) |
```
Comment (Qlas) [inline src/pdu_header.cpp:59]: why are some header items set by const and others not?

File: src/pdu_header.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+           (utils::toUnderlying(direction) << 3) | (utils::toUnderlying(transmissionMode) << 2) |
+           (utils::toUnderlying(crcFlag) << 1) | (utils::toUnderlying(largeFileFlag) << 0);
+}
+
+template <class EntityIDType, class SequenceNumberType>
+    requires std::unsigned_integral<EntityIDType> && std::unsigned_integral<SequenceNumberType>
+uint8_t cfdp::pdu::PduHeader<EntityIDType, SequenceNumberType>::buildThirdByte() const
```
Comment (Qlas) [inline src/pdu_header.cpp:66]: Are you sure that this is third byte?

PDU data field length is 16 bits

image

File: src/pdu_header.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+#include "utils.hpp"
+
+/*
+    Without the last 3 fields, the rest of the PDU header has constant
+    size. Calculating all used bits in order:
+    sizeBits = 3 + 1 + 1 + 1 + 1 + 1 + 16 + 1 + 3 + 1 + 3 = 32
+*/
+inline constexpr uint16_t CONST_HEADER_SIZE_BYTES      = 32 / 8;
+inline constexpr uint8_t VERSION_POSITION              = 5;
```
Comment (Qlas) [inline src/pdu_header.cpp:11]: I'm not sure if it's a good idea to set a const position from right side of a particular byte

File: src/utils.cpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#include "utils.hpp"
+
+inline constexpr uint8_t BITS_IN_BYTE = 8;
+
+template <class UnsignedInteger>
+    requires std::unsigned_integral<UnsignedInteger>
+std::vector<uint8_t> utils::intToBytes(UnsignedInteger value)
+{
+    // This cast is safe, the biggest we can get is 8.
```
Comment (Qlas) [inline src/utils.cpp:9]: Are you sure about that?
realPduDataFieldLength is uint16

File: src/pdu_header.cpp (hunk 2 above)
Comment (uncommon-nickname) [reply src/pdu_header.cpp:66]: ur right, mb

File: src/utils.cpp (hunk 4 above)
Comment (uncommon-nickname) [reply src/utils.cpp:9]: yeah, we can call it on u8 to u64

**Pull request merged**